### PR TITLE
Add swap-specific (a.k.a. swap conformance) jobs

### PR DESCRIPTION
**Pull request by iholder101**
In https://github.com/kubernetes/kubernetes/pull/123557 I'm trying to add some swap-related serial tests.

Although the tests are designed to be as lightweight as possible, swap by nature is aggressive since in order for it to kick in a memory pressure has to take place. This might hurt the node's stability and cause the current swap serial lanes to be flaky.

**Before this PR:**
We have two kinds of swap jobs:
- `node-swap-fedora-serial` and `node-swap-ubuntu-serial`:  Running all serial tests with swap enabled on the node.
- `node-swap-fedora`: Running all node conformance tests with swap enabled on the node.

Swap serial tests are running under swap serial jobs, which makes the swap serial jobs flaky.

**After this PR:**
In addition to the existing job, `node-swap-conformance-fedora-serial` and `node-swap-conformance-ubuntu-serial` are introduced.

Swap serial tests would run on these jobs to keep the previous jobs stable.

### CODE REVIEW

File: config/jobs/kubernetes/sig-node/node-kubelet.yaml [hunk 1]
```diff
@@ -414,6 +414,109 @@ periodics:
         - name: IGNITION_INJECT_GCE_SSH_PUBLIC_KEY_FILE
           value: "1"
 
+- name: ci-kubernetes-node-swap-conformance-ubuntu-serial
+  interval: 4h
+  cluster: k8s-infra-prow-build
+  annotations:
+    testgrid-dashboards: sig-node-kubelet
+    testgrid-tab-name: kubelet-gce-e2e-swap-conformance-ubuntu-serial
```
Comment (kannon92) [inline config/jobs/kubernetes/sig-node/node-kubelet.yaml:422]:
```suggestion
    testgrid-tab-name: kubelet-swap-conformance-ubuntu-serial
```

Comment (kannon92) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml:422]: nit mostly.

Comment (iholder101) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml:422]: Thanks @kannon92.

I was following the current conventions:
https://github.com/kubernetes/test-infra/blob/177cabeadc08c51f134bfdab398d71300531bf38/config/jobs/kubernetes/sig-node/node-kubelet.yaml#L365-L370

Do you think that in this case we should use something different?
Should we also change the testgrid names for existing swap serial jobs?

Comment (kannon92) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml:422]: https://github.com/kubernetes/test-infra/blob/177cabeadc08c51f134bfdab398d71300531bf38/config/jobs/kubernetes/sig-node/README.md

We don't really have a convention here. That link doesn't really mention gce but os, container runtime and serial.

Comment (kannon92) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml:422]: I'm fine to leave as is though.

Comment (iholder101) [reply config/jobs/kubernetes/sig-node/node-kubelet.yaml]: I like your testgrid name better, just wanted to make sure it's valid.

Thanks! Done :)

**Review by kannon92 — COMMENTED** [review]
/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**